projects/commudle-admin/src/app/feature-modules/events/components/user-event-registrations/entry-pass-scan/entry-pass-scan.component.ts

```diff
@@ -0,0 +1,92 @@
+import { Component, ElementRef, OnInit, ViewChild } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { BarcodeFormat } from '@zxing/library';
+import { ZXingScannerComponent } from '@zxing/ngx-scanner';
+import { EventPassService } from 'projects/commudle-admin/src/app/feature-modules/events/services/event-pass.service';
+import { IEventPass } from 'projects/shared-models/event-pass.model';
+import { LibToastLogService } from 'projects/shared-services/lib-toastlog.service';
+
+@Component({
+  selector: 'app-entry-pass-scan',
+  templateUrl: './entry-pass-scan.component.html',
+  styleUrls: ['./entry-pass-scan.component.scss'],
+})
+export class EntryPassScanComponent implements OnInit {
+
+  @ViewChild('scanner', { static: false }) scanner: ZXingScannerComponent;
+  @ViewChild('correctSound') correctSound: ElementRef;
+  @ViewChild('incorrectSound') incorrectSound: ElementRef;
+
+  scannerEnabled: boolean = true;
+
+  availableDevices: MediaDeviceInfo[];
+  currentDevice: MediaDeviceInfo = null;
+  hasDevices: boolean;
+  hasPermission: boolean;
+  qrResultString: string;
+  eventPass: IEventPass = null;
+  event_id: string = null;
+
+  formatsEnabled: BarcodeFormat[] = [
+    BarcodeFormat.QR_CODE,
+  ];
+
+  constructor(
+    private eventPassService: EventPassService,
+    private activatedRoute: ActivatedRoute,
+    private toastService: LibToastLogService,
+  ) { }
+
+  ngOnInit(): void {
+    this.activatedRoute.params.subscribe((res) => {
+      this.event_id = res.event_id;
+    })
+  }
+
+  isBoolean(val) {
```

Comment:
Let's just use `typeof` to check for boolean

Reply:
done!